packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts

```diff
@@ -0,0 +1,140 @@
+
+const MS_HIGHCONTRAST_ACTIVE = '@media screen and (-ms-high-contrast: active)';
+const MS_HIGHCONTRAST_BLACK_ON_WHITE = '@media screen and (-ms-high-contrast: black-on-white)';
+
+export const getStyles = memoizeFunction((
+  theme: ITheme,
+  customStyles?: ICheckboxStyles
+): ICheckboxStyles => {
+  const { semanticColors, palette } = theme;
+  const checkmarkFontColor = semanticColors.bodyBackground;
+  const checkboxBorderColor = semanticColors.inputBorder;
+  const checkboxHoveredBorderColor = semanticColors.inputBorderHovered;
+  const checkboxCheckedBackground = semanticColors.inputBackgroundChecked;
+  const checkboxCheckedHoveredBackground = semanticColors.inputBackgroundCheckedHovered;
+  const checkboxDisabledBackground = semanticColors.disabledText;
+  const checkboxDisabledTextColor = semanticColors.disabledText;
+
+  const styles: ICheckboxStyles = {
+    root: {
+      overflow: 'hidden',
+      position: 'relative',
+    },
+    label: [
+      getFocusStyleBasedOnAncestorClass(theme, '.ms-Fabric.is-focusVisible .is-inFocus'),
+      {
+        display: 'inline-flex',
+        alignItems: 'center',
+        cursor: 'pointer',
+        position: 'relative',
+        margin: '0 -4px',
+        userSelect: 'none'
+      }
+    ],
+    labelReversed: {
+      flexDirection: 'row-reverse',
+      justifyContent: 'flex-end'
+    },
+    labelDisabled: {
+      cursor: 'default'
+    },
+    input: {
+      position: 'absolute',
+      opacity: '0'
+    },
+    box: {
+      display: 'flex',
+      margin: '0 4px',
+      alignItems: 'center',
+      justifyContent: 'center',
+      flex: '0 0 ' + MS_CHECKBOX_LABEL_SIZE,
+      height: MS_CHECKBOX_LABEL_SIZE,
+    },
+    checkbox: {
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center',
+      height: MS_CHECKBOX_LABEL_SIZE,
+      width: MS_CHECKBOX_LABEL_SIZE,
+      borderWidth: '1px',
+      borderStyle: 'solid',
+      borderColor: checkboxBorderColor,
+      boxSizing: 'border-box',
+      background: 'none',
+      transitionProperty: 'background, border, border-color',
+      transitionDuration: MS_CHECKBOX_TRANSITION_DURATION,
+      transitionTimingFunction: MS_CHECKBOX_TRANSITION_TIMING,
+
+      /* incase the icon is bigger than the box */
+      overflow: 'hidden',
+    },
+    checkboxHovered: {
+      borderColor: checkboxHoveredBorderColor,
+    },
+    checkboxChecked: {
+      background: checkboxCheckedBackground,
+    },
+    checkboxCheckedHovered: {
+      background: checkboxCheckedHoveredBackground,
+    },
+    checkboxDisabled: {
+      background: checkboxDisabledBackground,
+      [MS_HIGHCONTRAST_ACTIVE]: {
+        borderColor: palette.contrastBlackDisabled,
+      },
+      [MS_HIGHCONTRAST_BLACK_ON_WHITE]: {
+        borderColor: palette.contrastWhiteDisabled,
+      },
+    },
+    checkboxCheckedDisabled: {
+      background: checkboxDisabledBackground,
+    },
+    checkmark: {
+      opacity: '0',

```

Comment:
Same as I mentioned above, you'll need checkmarkUnchecked to complement the checkmarkChecked, due to being unable to guarantee style order.

Additionally, I think you actually want to set it to width/height:0 initially (or maybe font-size 0, whichever works), and then in checkmarkChecked set it to what it should be. That way it'll animate the popping in effect. (Hopefully. If it doesn't work then nevermind.)